plugins/auto-sizes/optimization-detective.php

```diff
@@ -59,8 +59,12 @@ function auto_sizes_visit_tag( OD_Tag_Visitor_Context $context ): bool {
  * @param OD_Tag_Visitor_Registry $registry Tag visitor registry.
  */
 function auto_sizes_register_tag_visitors( OD_Tag_Visitor_Registry $registry ): void {
-	$registry->register( 'auto-sizes', 'auto_sizes_visit_tag' );
+	$registry->register(
+		'auto-sizes',
+		'auto_sizes_visit_tag',
+		defined( 'Image_Prioritizer_Img_Tag_Visitor::ID' ) ? array( Image_Prioritizer_Img_Tag_Visitor::ID => new Image_Prioritizer_Img_Tag_Visitor() ) : array()
```

Comment:
Only the ID should be passed here:

```suggestion
		defined( 'Image_Prioritizer_Img_Tag_Visitor::ID' ) ? array( Image_Prioritizer_Img_Tag_Visitor::ID ) : array()
```

Registering the `Image_Prioritizer_Img_Tag_Visitor` tag visitor is not the responsibility of the Enhanced Responsive Images plugin, but rather the Image Prioritizer plugin here:

https://github.com/WordPress/performance/blob/16b896d533fa865989f481333041669ab46b8986/plugins/image-prioritizer/helper.php#L32-L39